docfx/doxygen2markdown.cc

```diff
@@ -552,6 +552,9 @@ bool AppendIfProgramListing(std::ostream& os, MarkdownContext const& ctx,
   if (std::string_view{node.name()} != "programlisting") return false;
   // Start with a new paragraph, with the right level of indentation, and a new
   // code fence.
+  if (ctx.paragraph_start.empty()) {
```

Comment:
The obvious problem of always inserting newlines is what happens if we have a `<programlisting>` in the middle of a list.  We want to generate something like this (ignore the backslashes):

```md
- List Item 1
  \```cpp
  int Example() { return 42; }
  \```
- List Item 2
```

And not something like this:

```md
- List Item 1

  \```cpp
  int Example() { return 42; }
  \```
- List Item 2
```

Because in the second case the list is broken into two lists.

Keep in mind: we may not have any `<programlisting>` in the middle of a list (or anything else that requires not breaking paragraphs with two newlines).  So your fix may be perfectly fine now, but may break in the future.

Reply:
There might be a better fix, but when I tried to change the place where it was called (https://github.com/googleapis/google-cloud-cpp/blob/main/docfx/doxygen2yaml.cc#L105) and that is not covered by the XML unit tests and broke other tests.

I think the work to investigate isn't worth it.

Reply:
I suspect we need something like this:

https://github.com/googleapis/google-cloud-cpp/blob/9ea3fb29d82b7bed6c0c9f988019e293d1dd1ca9/docfx/doxygen2markdown.cc#L212-L213

Around here:

https://github.com/googleapis/google-cloud-cpp/blob/9ea3fb29d82b7bed6c0c9f988019e293d1dd1ca9/docfx/doxygen2markdown.cc#L528

That is, when a paragraph has multiple elements we need to reset the `paragraph_start` to `"\n\n"` (or to a single `"\n"`, not sure) after the first element.  

I don't want to have the perfect stop us from making progress here.

Reply:
Ack, this will be a consideration for the future